helper/schema/resource_data.go

```diff
@@ -59,6 +63,93 @@ type getResult struct {
 	Schema         *Schema
 }
 
+// TfTypeIdentity returns the identity data as a tftypes.Value.
```

Comment:
We should add some tests for these 🙂 , same comment for the other method.

It doesn't need to be as robust as the tests directly against `ToTfValue`, but at least testing the additional logic in this method (like adding attributes, or the flatmapping of data)